src/functions/op.js

```diff
@@ -132,48 +152,29 @@ const htmlBuilder = (group, options) => {
                 positionData: top,
                 children: [
                     {type: "kern", size: options.fontMetrics().bigOpSpacing5},
-                    {type: "elem", elem: subm, marginLeft: -slant + "em"},
-                    {type: "kern", size: subKern},
+                    {type: "elem", elem: sub.elem, marginLeft: -slant + "em"},
+                    {type: "kern", size: sub.kern},
                     {type: "elem", elem: base},
                 ],
             }, options);
-        } else if (!subGroup) {
-            bottom = base.depth + baseShift;
+        } else if (sup) {
+            const bottom = base.depth + baseShift;
 
             finalGroup = buildCommon.makeVList({
                 positionType: "bottom",
                 positionData: bottom,
                 children: [
                     {type: "elem", elem: base},
-                    {type: "kern", size: supKern},
-                    {type: "elem", elem: supm, marginLeft: slant + "em"},
+                    {type: "kern", size: sup.kern},
+                    {type: "elem", elem: sup.elem, marginLeft: slant + "em"},
                     {type: "kern", size: options.fontMetrics().bigOpSpacing5},
                 ],
             }, options);
-        } else if (!supGroup && !subGroup) {
+        } else {
             // This case probably shouldn't occur (this would mean the
             // supsub was sending us a group with no superscript or
             // subscript) but be safe.
             return base;
-        } else {
-            bottom = options.fontMetrics().bigOpSpacing5 +
```

Comment:
I moved this to the top to make all the if statements positive so that type refinement could guarantee what was non-nullable in each if-then statement body.